### PR TITLE
[6.2.z] fix image not destroyed at provisioning server when using custom hostname

### PR DESCRIPTION
**Pull request by ldjebran**
issue: https://github.com/SatelliteQE/robottelo/issues/4372
The main idea is to not rewrite the initialization arguments, as they are needed in all virtual machine life usage.

python console log without hostname: http://pastebin.test.redhat.com/457666
python console log with hostname: http://pastebin.test.redhat.com/457667

```console
(sat-6.2.z) dlezz@elysion:~/projects/robottelo-fork$ py.test tests/foreman/cli/test_host.py -v -k "KatelloAgentTestCase" 
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-2.9.2, py-1.4.32, pluggy-0.3.1 -- /home/dlezz/.pyenv/versions/sat-6.2.z/bin/python
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile: 
plugins: cov-2.3.1
collected 40 items 
2017-02-21 19:18:06 - conftest - DEBUG - Found WONTFIX in decorated tests ['1269196', '1103157', '1156555', '1230902', '1245334', '1204686', '1267224', '1226425', '1217635', '1079482', '1214312']

2017-02-21 19:18:06 - conftest - DEBUG - Collected 40 test cases


tests/foreman/cli/test_host.py::KatelloAgentTestCase::test_negative_unregister_and_pull_content PASSED
tests/foreman/cli/test_host.py::KatelloAgentTestCase::test_positive_apply_errata <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_host.py::KatelloAgentTestCase::test_positive_get_errata_info <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_host.py::KatelloAgentTestCase::test_positive_install_package <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_host.py::KatelloAgentTestCase::test_positive_install_package_group <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_host.py::KatelloAgentTestCase::test_positive_remove_package <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_host.py::KatelloAgentTestCase::test_positive_remove_package_group <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_host.py::KatelloAgentTestCase::test_positive_upgrade_package <- robottelo/decorators/__init__.py PASSED
tests/foreman/cli/test_host.py::KatelloAgentTestCase::test_positive_upgrade_packages_all <- robottelo/decorators/__init__.py PASSED

=================================== 31 tests deselected by '-kKatelloAgentTestCase' ====================================
====================================== 9 passed, 31 deselected in 2198.98 seconds ======================================
(sat-6.2.z) dlezz@elysion:~/projects/robottelo-fork$ 
```
**to be cherry-picked to 6.3**

### CODE REVIEW

File: robottelo/vm.py [hunk 1]
```diff
@@ -89,6 +89,33 @@ def __init__(
...
+    def hostname(self):
+        if self._hostname:
+            return self._hostname
+        else:
+            return u'{0}.{1}'.format(self._target_image, self.domain)
+
+    @property
+    def target_image(self):
+        if self._hostname:
+            return self._target_image
```
Comment (rochacbruno) [inline robottelo/vm.py]: `self._target_image` is set only if `tag` is present. shouldn't be set default to None? (see other comment) or testes here `if self.tag and self._hostname` ?

File: robottelo/vm.py [hunk 2]
```diff
@@ -89,6 +89,33 @@ def __init__(
         if tag:
             self._target_image = tag + self._target_image
```
Comment (rochacbruno) [inline robottelo/vm.py]: Maybe

```python
self._target_image = (tag + self._target_image) if tag else None
```

So we ensure `_target_image` attribute always exists with a default value to None?

Comment (ldjebran) [inline robottelo/vm.py]: @rochacbruno , please refer to my upper comment

File: robottelo/vm.py (hunk 1 above)
Comment (ldjebran) [inline robottelo/vm.py]: @rochacbruno , self._target_image  is already defined at line 88 one line upper if tag in  __init__ function
```python
    self._target_image = target_image or str(id(self))
```
it's perhaps hidden in github , you need to expend that section

**Comment by omaciel** [comment]
```
13:54:18 omaciel: dlezz++
13:54:18 robottelo: dlezz has 22 points
```

**Comment by ldjebran** [comment]
@omaciel, many thanks

**Pull request merged**